src/main/java/net/snowflake/client/core/arrow/ThreeFieldStructToTimestampTZConverter.java

```diff
@@ -138,4 +129,30 @@ public short toShort(int rowIndex) throws SFException {
     throw new SFException(
         ErrorCode.INVALID_VALUE_CONVERT, logicalTypeStr, SnowflakeUtil.SHORT_STR, "");
   }
+
+  public static Timestamp getTimestamp(
+      long epoch,
+      int fraction,
+      int timeZoneIndex,
+      boolean fromToString,
+      long resultVersion,
+      boolean useSessionTimezone)
+      throws SFException {
+    if (ArrowResultUtil.isTimestampOverflow(epoch)) {
+      if (fromToString) {
+        throw new TimestampOperationNotAvailableException(epoch, fraction);
+      } else {
+        return null;
+      }
+    }
+
+    TimeZone timeZone;
+    if (resultVersion > 0) {
+      timeZone = SFTimestamp.convertTimezoneIndexToTimeZone(timeZoneIndex);
+    } else {
+      timeZone = TimeZone.getTimeZone("UTC");
```

Comment:
Is it a correct assumption?

Reply:
This code was only extracted to static method, but I think it makes sense to make the timezone without any offset when the index is not present.